dialer.go

```diff
@@ -34,7 +37,7 @@ import (
 // serverPublicKey - if provided, this dialer will use encryption.
 //
 // dial - function to open an underlying connection.
-func NewDialer(windowSize int, maxPadding int, maxStreamsPerConn uint16, pingInterval time.Duration, pool BufferPool, cipherCode Cipher, serverPublicKey *rsa.PublicKey, dial func() (net.Conn, error)) Dialer {
+func NewDialer(windowSize int, maxPadding int, maxStreamsPerConn uint16, idleInterval time.Duration, pingInterval time.Duration, pool BufferPool, cipherCode Cipher, serverPublicKey *rsa.PublicKey) Dialer {
```

Comment:
It is non-trivial to check if the parameters are passed correctly and in the correct order. Consider passing them as a struct?